pandas/core/arrays/datetimelike.py

```diff
@@ -1303,6 +1303,9 @@ def __rsub__(self, other):
         if is_datetime64_any_dtype(other) and is_timedelta64_dtype(self.dtype):
             # ndarray[datetime64] cannot be subtracted from self, so
             # we need to wrap in DatetimeArray/Index and flip the operation
+            if lib.is_scalar(other):
+                # i.e. np.datetime64 object
```

Comment:
null hit this path too? (np.datetime64(NaT))

Reply:
i think so, yah